lighthouse-core/computed/computed-artifact.js

```diff
@@ -11,28 +11,35 @@ const log = require('lighthouse-logger');
 /**
  * Decorate computableArtifact with a caching `request()` method which will
  * automatically call `computableArtifact.compute_()` under the hood.
- * @template {{name: string, compute_(artifacts: unknown, context: LH.Artifacts.ComputedContext): Promise<unknown>}} C
+ * @template {{name: string, compute_(dependencies: unknown, context: LH.Artifacts.ComputedContext): Promise<unknown>}} C
+ * @template {Array<keyof FirstParamType<C['compute_']>>} K
  * @param {C} computableArtifact
+ * @param {(K & ([keyof FirstParamType<C['compute_']>] extends [K[number]] ? unknown : never)) | null} keys List of dependencies that will be checked for equality when caching. Use `null` to check all dependencies.
```

Comment:
maybe something like:

```suggestion
 * @param {(K & ([keyof FirstParamType<C['compute_']>] extends [K[number]] ? unknown : never)) | null} keys List of properties of `dependencies` used by `compute_`; other properties are filtered out. Use `null` to allow all properties. Ensures that only required properties are used for caching result.
```
otherwise it sounds a bit like you could mess up caching if you don't cache on a property but you still use it in the computed artifact